tests/unit/bokeh/core/property/test_container.py

```diff
@@ -144,6 +144,29 @@ def test_init(self) -> None:
         with pytest.raises(TypeError):
             bcpc.Dict()
 
+    def test_validate_invalid_key(self) -> None:
+        d = bcpc.Dict(String, String)
+        with pytest.raises(ValueError) as err:
+            d.validate({"Foo": "Bar", 1: "Baz"})
+            assert("invalid keys: 1" in err.value.args[0])
+
+    def test_validate_invalid_value(self) -> None:
+        d = bcpc.Dict(String, String)
+        with pytest.raises(ValueError) as err:
+            d.validate({"Foo": "Bar", "Baz": 1})
+            assert("values for keys: Baz" in err.value.args[0])
+
+    def test_validate_invalid_arg(self) -> None:
+        d = bcpc.Dict(String, String)
+        with pytest.raises(ValueError) as err:
+            d.validate([])
+            assert("got a value of type list" in err.value.args[0])
+
+    def test_validate_valid(self) -> None:
+        d = bcpc.Dict(String, String)
+        d.validate({"Foo": "Bar"})
+        assert True
```

Comment:
I like having positive tests, but the actual check here is that no exception is raised. I think you can use `with pyest. does_not_raise():` for that.

Reply:
That function doesn't seem to exist

Reply:
I guess the first refs I found skipped some details, should have just gone straight to the docs:

https://docs.pytest.org/en/7.1.x/example/parametrize.html#parametrizing-conditional-raising

TLDR, it's actually: `from contextlib import nullcontext as does_not_raise`